src/src/rtc6705.c

```diff
@@ -7,19 +7,68 @@
 static gpio_out_t ss_pin;
 static gpio_out_t sck_pin;
 static gpio_out_t mosi_pin;
-static uint_fast8_t amp_state;
+static gpio_in_t miso_pin;
 
 static uint32_t powerUpAfterSettleTime = 0;
 
-void spiPinSetup(void)
+void rtc6705spiPinSetup(void)
 {
   ss_pin = gpio_out_setup(SPI_SS, 1);
   sck_pin = gpio_out_setup(SPI_CLOCK, 0);
   mosi_pin = gpio_out_setup(SPI_MOSI, 0);
-  amp_state = 0;
 }
 
-void sendBits(uint32_t data)
+uint32_t rtc6705readRegister(uint8_t readRegister)
+{
+  uint32_t writeData = readRegister | (READ_BIT << 4);
+  uint32_t readData = 0;
+
+  gpio_out_write(ss_pin, 0);
+  delayMicroseconds(1);
+
+  // Write register address and read bit
+  for (uint8_t i = 0; i < 5; i++)
+  {
+    gpio_out_write(sck_pin, 0);
+    delayMicroseconds(1);
+    gpio_out_write(mosi_pin, writeData & 0x1);
+    delayMicroseconds(1);
+    gpio_out_write(sck_pin, 1);
+    delayMicroseconds(1);
+
+    writeData >>= 1;
+  }
+
+  // Change pin from output to input
+  miso_pin = gpio_in_setup(SPI_MOSI, 0);
```

Comment:
remove global `static gpio_in_t miso_pin;` and move it to local here since it is not needed outside of this method. That way is will be much faster.
==> `gpio_in_t miso_pin = gpio_in_setup(SPI_MOSI, 0);`